src/components/Oauth/WorkspaceEntry/WorkspaceOauthFlow.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * OAuth flow for any providers that do not require the consumer to enter a subdomain first.
+ */
+
+import { useCallback, useState } from 'react';
+
+import { useApiKey } from '../../../context/ApiKeyContextProvider';
+import { useProject } from '../../../context/ProjectContextProvider';
+import { capitalize } from '../../../utils';
+import { fetchOAuthCallbackURL } from '../fetchOAuthCallbackURL';
+import OAuthPopup from '../OAuthPopup';
+
+import { WorkspaceEntry } from './WorkspaceEntry';
+
+interface NoSubdomainOauthFlowProps {
+  provider: string;
+  consumerRef: string;
+  consumerName?: string;
+  groupRef: string;
+  groupName?: string;
+}
+
+/**
+ * WorkspaceEntry is generic for any provider that requires a workspace to be entered first,
+ * then launches a popup with the OAuth flow.
+ */
+export function WorkspaceOauthFlow({
+  provider, consumerRef, consumerName, groupRef, groupName,
+}: NoSubdomainOauthFlowProps) {
+  const { projectId } = useProject();
+  const apiKey = useApiKey();
+
+  const [workspace, setWorkspace] = useState<string>('');
+  const [oAuthCallbackURL, setOAuthCallbackURL] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  //  fetch OAuth callback URL from connection so that oath popup can be launched
+  const handleSubmit = async () => {
+    setError(null);
```

Comment:
Can you add an if statement to check that `workspace` is not an empty string, if so, call `setError('Workspace name cannot be empty)`

Reply:
this shouldn't be possible since the submit button is disabled when the workspace is empty, but I can add in this check in the final polish pr.